src/els_methods.erl

```diff
@@ -117,12 +113,26 @@ method_to_function_name(Method) ->
   Binary   = els_utils:to_binary(Lower),
   binary_to_atom(Binary, utf8).
 
+%%==============================================================================
+%% Initialize
+%%==============================================================================
+
+-spec initialize(params(), state()) -> result().
+initialize(Params, State) ->
+  Provider = els_general_provider,
+  Request  = {initialize, Params},
+  Response = els_provider:handle_request(Provider, Request),
+  {response, Response, State#{status => initialized}}.
+
 %%==============================================================================
 %% Initialized
 %%==============================================================================
 
 -spec initialized(params(), state()) -> result().
-initialized(_Params, State) ->
+initialized(Params, State) ->
+  Provider = els_general_provider,
+  Request  = {initialized, Params},
+  _Response = els_provider:handle_request(Provider, Request),
   %% Report to the user the server version
```

Comment:
Why not move all of this to the `els_general_provider` module?

Reply:
Because I wanted to refactor this separately, for two main reasons:

1. In `3.15`, the initiate method allows replying with `ServerInfo`, which include exactly name and version. We should not send them back explicitly as a "showMessage" because this is not part of the protocol.
2. By moving the implementation to the provider, the `Methods` and `Params` should be returned in a single value, so it can be used transparently by this module.

I will create an issue with this.